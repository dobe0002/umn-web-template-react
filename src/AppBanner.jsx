import React from 'react';
import styled from 'styled-components';
import PropTypes from 'prop-types';

const HeaderWrapper = styled.div`
  padding-left: 10px;
  padding-top: 8px;
  min-height: ${60 - 8}px;
  border-bottom: #f93 2px solid;
  font-family: 'Trebuchet MS', Arial, Helvetica, sans-serif;
  font-size: 35px;
  font-weight: bold;
  color: #5b0013;
  margin-top: 0;
  margin-bottom: 0;
  background: #f0e9d1; /* Old browsers */
  background: -moz-linear-gradient(
    top,
    #f0e9d1 0%,
    #ffde7c 100%
  ); /* FF3.6-15 */
  background: -webkit-linear-gradient(
    top,
    #f0e9d1 0%,
    #ffde7c 100%
  ); /* Chrome10-25,Safari5.1-6 */
  background: linear-gradient(
    to bottom,
    #f0e9d1 0%,
    #ffde7c 100%
  ); /* W3C, IE10+, FF16+, Chrome26+, Opera12+, Safari7+ */
`;
const AppBanner = props => {
  const { appTitle } = props;
  return <HeaderWrapper role='contentinfo'>{appTitle}</HeaderWrapper>;
};
AppBanner.propTypes = {
  appTitle: PropTypes.string.isRequired
};

export default AppBanner;
