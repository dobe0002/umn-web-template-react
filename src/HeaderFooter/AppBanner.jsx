/* eslint-disable react/prop-types */
/* eslint-disable no-dupe-keys */
import React from 'react';
// import styled from 'styled-components'; /* Not using styled-components so  styled-components isn't needed as a dependency */
// import PropTypes from 'prop-types';  /* Disabling this so prop-types isn't needed as a dependency */

const bannerCSS = {
  paddingLeft: '10px',
  paddingTop: '8px',
  minHeight: '52px',
  borderBottom: ' #f93 2px solid',
  fontFamily: '"Trebuchet MS", Arial, Helvetica, sans-serif',
  fontSize: '35px',
  fontWeight: 'bold',
  color: '#5b0013',
  marginTop: 0,
  marginBottom: 0,
  background: '#f0e9d1' /* Old browsers */,
  background:
    '-moz-linear-gradient( top, #f0e9d1 0%, #ffde7c 100%)' /* FF3.6-15 */,
  background:
    '-webkit-linear-gradient( top, #f0e9d1 0%, #ffde7c 100%)' /* Chrome10-25,Safari5.1-6 */,
  background:
    'linear-gradient( to bottom, #f0e9d1 0%,#ffde7c 100%)' /* W3C, IE10+, FF16+, Chrome26+, Opera12+, Safari7+ */,
};
const AppBanner = (props) => {
  const { appTitle } = props;
  return (
    <div role='contentinfo' style={bannerCSS}>
      {appTitle}
    </div>
  );
};
// AppBanner.propTypes = {
//   appTitle: PropTypes.string.isRequired,
// };

export default AppBanner;
