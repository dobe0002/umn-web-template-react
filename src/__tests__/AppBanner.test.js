/* eslint-disable react/jsx-filename-extension */
import React from 'react';
import { configure, shallow, mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import AppBanner from '../HeaderFooter/AppBanner';

const { axe, toHaveNoViolations } = require('jest-axe');

expect.extend(toHaveNoViolations);
configure({ adapter: new Adapter() });

describe('Application Banner', () => {
  it('Accessibility check', async () => {
    const appBanner = mount(<AppBanner appTitle='My super duper app' />);
    const results = await axe(appBanner.getDOMNode());

    // console.log('Axe violations', results.violations);

    expect(results).toHaveNoViolations();
  });
  it('AppBanner renders correctly', () => {
    const appBanner = shallow(<AppBanner appTitle='My super duper app' />);
    expect(appBanner).toMatchSnapshot();
  });
});
