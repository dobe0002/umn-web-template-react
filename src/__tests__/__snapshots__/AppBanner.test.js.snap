// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Application Banner AppBanner renders correctly 1`] = `
<div
  role="contentinfo"
  style={
    Object {
      "background": "linear-gradient( to bottom, #f0e9d1 0%,#ffde7c 100%)",
      "borderBottom": " #f93 2px solid",
      "color": "#5b0013",
      "fontFamily": "\\"Trebuchet MS\\", Arial, Helvetica, sans-serif",
      "fontSize": "35px",
      "fontWeight": "bold",
      "marginBottom": 0,
      "marginTop": 0,
      "minHeight": "52px",
      "paddingLeft": "10px",
      "paddingTop": "8px",
    }
  }
>
  My super duper app
</div>
`;
